bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/inst/OiwfsPlotFeature.scala

```diff
@@ -0,0 +1,195 @@
+package jsky.app.ot.gemini.inst
+
+import edu.gemini.pot.ModelConverters._
+import edu.gemini.spModel.core._
+import edu.gemini.spModel.core.AngleSyntax._
+import edu.gemini.spModel.gemini.flamingos2.{F2OiwfsProbeArm, Flamingos2OiwfsGuideProbe}
+import edu.gemini.spModel.gemini.gmos.{GmosSouthOiwfsProbeArm, GmosNorthOiwfsProbeArm, GmosOiwfsGuideProbe}
+import edu.gemini.spModel.guide.{PatrolField, OffsetValidatingGuideProbe}
+import edu.gemini.spModel.inst.{FeatureGeometry, ProbeArmGeometry}
+import edu.gemini.spModel.obs.context.ObsContext
+import edu.gemini.spModel.rich.shared.immutable.{asScalaOpt, asGeminiOpt}
+import edu.gemini.spModel.target.env.TargetEnvironmentDiff
+import edu.gemini.shared.util.immutable.{Option => JOption}
+import jsky.app.ot.tpe.TpeImageFeature.Figure
+
+import jsky.app.ot.tpe._
+import jsky.app.ot.util.PropertyWatcher
+
+import java.awt.{List => _, _}
+import java.awt.BasicStroke.{CAP_BUTT, JOIN_BEVEL}
+import java.beans.{PropertyChangeEvent, PropertyChangeListener}
+import java.util.logging.Logger
+
+import scala.collection.JavaConverters._
+import scalaz._
+import Scalaz._
+
+object OiwfsPlotFeature {
+  private val Log = Logger.getLogger(getClass.getName)
+
+  // Drawing specifications.
+  val ProbeArmColor        = Color.red
+  val ProbeArmStroke       = new BasicStroke(2F)
+  val Blocked              = AlphaComposite.getInstance(AlphaComposite.SRC_OVER, 0.5F)
+
+  // A TPE warning if there is no valid region.
+  val NoValidRegionWarning = TpeMessage.warningMessage("No valid OIWFS region. Check offset positions.")
+
+  // The color to use to draw the patrol field
+  val ReachableColor       = Color.green
+  val PatrolFieldColor     = Color.red
+
+  // Used to draw dashed lines
+  val DashedLineStroke  = new BasicStroke(0.5F, CAP_BUTT,  JOIN_BEVEL, 0F, Array[Float](12F, 12F), 0F)
+  val ThickDashedStroke = new BasicStroke(2.0F, CAP_BUTT,  JOIN_BEVEL, 0F, Array[Float](12F, 12F), 0F)
+
+  def fillObscuredArea: Boolean =
+    OIWFS_Feature.getProps.getBoolean(OIWFS_Feature.PROP_FILL_OBSCURED, true)
+}
+
+sealed class OiwfsPlotFeature(probe: OffsetValidatingGuideProbe, probeArm: ProbeArmGeometry)
+  extends TpeImageFeature(probe.getKey, s"Show the ${probe.getKey} patrol field and arm.") with PropertyWatcher {
+
+  OIWFS_Feature.getProps.addWatcher(this)
+
+  import OiwfsPlotFeature._
+  import FeatureGeometry._
+
+  def getFigures(tpeCtx: TpeContext): List[Figure] = {
+    def go(obsCtx: ObsContext, patField: PatrolField): List[Figure] = {
+      // Extract info from the TPE context, converted to new model.
+      val posAngle = Angle.fromDegrees(tpeCtx.instrument.posAngleOrZero)
+      val selPos   = tpeCtx.offsets.selectedPos
+      val offset   = selPos.map { opb =>
+        Offset(opb.getXaxis.arcsecs[OffsetP], opb.getYaxis.arcsecs[OffsetQ])
+      } | Offset.zero
```

Comment:
You can just say `.foldMap(...)` and remove the `| Offset.zero` if you define `Monoid[Offset]`, which is probably worthwhile at some point.